File: frontend/src/app/responses/bangxephang/weekly-rank-reward-response.ts
export interface WeeklyRankRewardResponse {
  gold_reward: number;
  rank_tier: 'BRONZE' | 'SILVER' | 'GOLD' | 'PLATINUM' | 'DIAMOND' | 'MASTER';
  global_rank: number | null;
  week_id: string;      // ví dụ "2025-51"
  claimed_before: boolean;
  gold_before: number;
  gold_after: number;
}

File: backend/src/main/java/com/app/backend/responses/bangxephang/WeeklyRankRewardResponse.java
package com.app.backend.responses.bangxephang;

import com.app.backend.models.constant.RankTier;
import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.*;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class WeeklyRankRewardResponse {

    @JsonProperty("gold_reward")
    private Long goldReward;

    @JsonProperty("rank_tier")
    private RankTier rankTier;

    @JsonProperty("global_rank")
    private Integer globalRank;

    @JsonProperty("week_id")
    private String weekId;      // ví dụ "2025-51"

    @JsonProperty("claimed_before")
    private boolean claimedBefore;

    @JsonProperty("gold_before")
    private Long goldBefore;

    @JsonProperty("gold_after")
    private Long goldAfter;
}
